pkg/resource/deploy/snapshot.go

```diff
@@ -155,6 +166,18 @@ func (snap *Snapshot) VerifyIntegrity() error {
 					}
 					return fmt.Errorf("child resource %s refers to missing parent %s", urn, par)
 				}
+
+				// Ensure that our URN is a child of the parent's URN.
+				expectedType := urn.Type()
+				if par.QualifiedType() != resource.RootStackType {
+					expectedType = par.QualifiedType() + "$" + expectedType
+				}
+
+				if urn.QualifiedType() != expectedType {
+					logging.Warningf("child resource %s has parent %s but it's URN doesn't match", urn, par)
+					// TODO: Change this to an error once we're sure users won't hit this in the wild.
+					// return fmt.Errorf("child resource %s has parent %s but it's URN doesn't match", urn, par)
```

Comment:
When will we know that? Can we open a tracking issue and link to it here? (In general, any TODO we add should link to an issue).

Reply:
Happy for this to be addressed in a subsequent change.

Reply:
> When will we know that? Can we open a tracking issue and link to it here? (In general, any TODO we add should link to an issue).

It should only be for deployments that failed, so rare. But I wanted to get at least one release out with this new behavior and us confident that its stable before making this an actual error.

Reply:
@justinvp @Frassle  Did we file a follow up issue for this? I don't love leaving what should be an error as a warning that people will ignore and which almost certainly represents a ticking time bomb ...